Plugins/Mute/Plugin.cs

```diff
@@ -7,97 +7,203 @@ namespace Mute;
+                        gameEvent.Origin.Tell(Utilities.CurrentLocalization
+                            .LocalizationIndex["PLUGINS_MUTE_REMAINING_TIME"].FormatExt(
+                                muteMetaSay.Expiration is not null
+                                    ? muteMetaSay.Expiration.Value.HumanizeForCurrentCulture()
+                                    : Utilities.CurrentLocalization.LocalizationIndex["PLUGINS_MUTE_NEVER"],
+                                muteMetaSay.Reason));
+                        break;
+                    case MuteState.Unmuting:
+                    case MuteState.Unmuted:
+                    default:
+                        break;
+                }
+
+                break;
+            case GameEvent.EventType.Update:
+                // Get correct EFClient object
+                var client = server.GetClientsAsList().FirstOrDefault(x => x.NetworkId == gameEvent.Origin.NetworkId);
+                if (client == null) break;
+
+                // Check if user has any meta set, else ignore (unmuted)
+                var muteMetaUpdate = await MuteManager.GetCurrentMuteState(client);
+                switch (muteMetaUpdate.MuteState)
+                {
+                    case MuteState.Muted:
+                        if (!muteMetaUpdate.CommandExecuted)
+                        {
+                            await server.ExecuteCommandAsync($"muteClient {client.ClientNumber}");
+                            muteMetaUpdate.CommandExecuted = true;
+                        }
+
+                        await IsExpiredMute(client, server);
                         break;
                     case MuteState.Unmuting:
-                        await server.ExecuteCommandAsync($"unmute {gameEvent.Origin.ClientNumber}");
-                        await DataManager.WritePersistentData(gameEvent.Origin, MuteState.Unmuted);
                         break;
                     case MuteState.Unmuted:
+                        if (!muteMetaUpdate.CommandExecuted)
+                        {
+                            await server.ExecuteCommandAsync($"unmute {client.ClientNumber}");
+                            muteMetaUpdate.CommandExecuted = true;
+                        }
+
                         break;
                 }
 
                 break;
         }
     }
 
-    public Task OnLoadAsync(IManager manager)
+    private async Task<bool> IsExpiredMute(EFClient client, Server server)
     {
-        _interactionRegistration.RegisterInteraction(MuteInteraction, async (clientId, game, token) =>
+        var expiration = (await MuteManager.GetCurrentMuteState(client)).Expiration;
+
+        // Return if permanent mute and not yet expired
+        if (expiration is null) return false;
+        if (!(expiration < DateTime.UtcNow)) return false;
+
+        await MuteManager.Unmute(server, Utilities.IW4MAdminClient(), client,
+            Utilities.CurrentLocalization.LocalizationIndex["PLUGINS_MUTE_EXPIRED"]);
```

Comment:
Would recommend using the result of the call to an is expired method to then call the unmute method instead of doing it in the same place